TelegramNotifier

MiguelCarranza opened the pull request:
Simple [Telegram Bot](https://core.telegram.org/bots/api) Notifier

martomi commented on config-example.yaml:
```diff
@@ -14,3 +14,6 @@ notifier:
     user_key: 'dummy_key'
+  telegram:
+    bot_token: 'dummy_bot_token'
+    chat_id: 'dummy_chat_id'
```
Please add `enable: false` config key.

MiguelCarranza replied:
dang, I probably removed it when renaming the keys

martomi reviewed (changes requested):
@MiguelCarranza looks good, thanks for the PR! Unfortunately I forgot to mention in CONTRIBUTION.md that PRs should be opened against the `dev` branch. I changed the target now - could you please rebase?

greimela commented on src/notifier/telegram_notifier.py:
```diff
@@ -0,0 +1,45 @@
+                    {
+                        "chat_id": self.chat_id,
+                        "text": f"*{self._title_prefix} {event.service.name}*\n{event.message}",
+                        "parse_mode": "Markdown",
```
We could disable the notification in case of a low event priority.
```suggestion
                        "parse_mode": "Markdown",
                        "disable_notification": event.priority == EventPriority.LOW,
```

greimela commented on src/notifier/telegram_notifier.py:
```diff
@@ -0,0 +1,45 @@
+                request_body = json.dumps(
+                    {
+                        "chat_id": self.chat_id,
+                        "text": f"*{self._title_prefix} {event.service.name}*\n{event.message}",
```
I would add a 🚨 symbol in front of the text in case of a HIGH event priority, since Telegram does not support sending high priority messages itself.

martomi replied:
Good idea!

greimela commented on tests/notifier/test_telegram_notifier.py:
```diff
@@ -0,0 +1,37 @@
+from src.notifier.telegram_notifier import TelegramNotifier
+
+
+class TestTelegramNotifier(unittest.TestCase):
```
I would add some tests for the HIGH and NORMAL priority cases, if they get handled differently.

greimela reviewed (commented):
Oh, I also created a TelegramNotifier yesterday evening, it looks quite similar 😄 

I will add some suggestions here, for things I added in my version.

MiguelCarranza commented:
> @MiguelCarranza looks good, thanks for the PR! Unfortunately I forgot to mention in CONTRIBUTION.md that PRs should be opened against the `dev` branch. I changed the target now - could you please rebase?

Comments addressed and rebased!

martomi commented:
Perfect thanks! Only nitpick I have left is to squash the commits to keep history cleaner, but I probably can also do it after rebasing a bit later today :-)

greimela commented:
@martomi You could just use squash as merge strategy, I like to use that in my projects as well.

martomi commented:
@greimela Thanks, I'll try that but I don't like that there's little control over the resulting commit message 😄

greimela commented:
AFAIK you can freely edit the resulting commit message as maintainer. Per default it uses the pull request title, but you can modify it before merging.

martomi commented:
Wow, that's very cool, thanks for the tip!

martomi commented:
@greimela did you also use Telegram Bot for your implementation? I'm updating the README now and trying to understand how do you ensure that the bot cannot be subscribed by a third person?
